### PR TITLE
Fix for bug in identifyForm with empty forms.

### LINKED ISSUE #1072: identifyForm mishandles empty forms
Looks like identifyForm simply removes fields of a form to make it missing. However, for already-empty forms, that's a noop, so they wrongly pass validation successfully.

The usecase seems pretty legitimate to me, e.g. delete actions on a particular route (which need no parameters). Of course, one can always work around it by providing a different route and handler for each form.

### PR DESCRIPTION
**Pull request by jprider63**
Fixes #1072

### CODE REVIEW

File: yesod-form/Yesod/Form/Functions.hs [hunk 1]
```diff
@@ -336,21 +336,24 @@ identifyForm identVal form = \fragment -> do
...
     -- Run the form proper (with our hidden <input>).  If the
     -- data is missing, then do not provide any params to the
     -- form, which will turn its result into FormMissing.  Also,
     -- doing this avoids having lots of fields with red errors.
     let eraseParams | missing   = local (\(_, h, l) -> (Nothing, h, l))
                     | otherwise = id
-    eraseParams (form fragment')
+    ( res', w) <- eraseParams (form fragment')
+
+    let res = if missing then FormMissing else res'
```
Comment (snoyberg) [inline yesod-form/Yesod/Form/Functions.hs]: To ensure this doesn't get reverted in the future: can you add comment linking to the issue?

File: yesod-form/Yesod/Form/Functions.hs [hunk 2]
```diff
@@ -62,7 +62,7 @@ import Yesod.Core
 import Yesod.Core.Handler (defaultCsrfParamName)
```
Comment (snoyberg) [inline yesod-form/Yesod/Form/Functions.hs]: Can you provide a patch version bump in the cabal file, and add a ChangeLog entry about the change?

**Comment by snoyberg** [comment]
Looks good, just some housekeeping requests. Thanks!

**Comment by jprider63** [comment]
Done. Let me know if I should change anything else.

**Comment by snoyberg** [comment]
Thanks!

**Comment by snoyberg** [comment]
Thanks!

**Pull request merged**